diff -ruN untouched/gcc-11.2.0/gcc/config/i386/t-x86_64-elf gcc-11.2.0/gcc/config/i386/t-x86_64-elf
--- untouched/gcc-11.2.0/gcc/config/i386/t-x86_64-elf	1969-12-31 16:00:00.000000000 -0800
+++ gcc-11.2.0/gcc/config/i386/t-x86_64-elf	2022-02-21 20:52:57.566486400 -0800
@@ -0,0 +1,2 @@
+MULTILIB_OPTIONS += mno-red-zone
+MULTILIB_DIRNAMES += no-red-zone
\ No newline at end of file
diff -ruN untouched/gcc-11.2.0/gcc/config.gcc gcc-11.2.0/gcc/config.gcc
--- untouched/gcc-11.2.0/gcc/config.gcc	2021-07-27 23:55:07.000283400 -0700
+++ gcc-11.2.0/gcc/config.gcc	2022-02-21 20:53:36.139266900 -0800
@@ -1897,6 +1897,7 @@
 	tm_file="${tm_file} i386/unix.h i386/att.h dbxelf.h elfos.h newlib-stdint.h i386/i386elf.h"
 	;;
 x86_64-*-elf*)
+	tmake_file="${tmake_file} i386/t-x86_64-elf" # Add multilib configuration with no red zone
 	tm_file="${tm_file} i386/unix.h i386/att.h dbxelf.h elfos.h newlib-stdint.h i386/i386elf.h i386/x86-64.h"
 	;;
 x86_64-*-rtems*)
 